### ISSUE TITLE
macOS: no matching function for call to 'hasPrefix'

### ISSUE THREAD

**Issue by nmattia** [NixOS/nix#3240]
I'm on latest master (`d12d69ea1a871d631d77c8ef5e8468b4a2bff80f`). I've run the following:

```
$ nix-shell
$ ./bootstrap.sh
$ configurePhase
$ make
  CXX    src/libmain/shared.o
src/libmain/shared.cc:158:26: error: no matching function for call to 'hasPrefix'
    if (getuid() == 0 && hasPrefix(getEnv("TMPDIR"), "/var/folders/"))
                         ^~~~~~~~~
src/libutil/util.hh:406:6: note: candidate function not viable: no known conversion from
      'std::optional<std::string>' (aka 'optional<basic_string<char, char_traits<char>, allocator<char>
      > >') to 'const std::__1::string' (aka 'const basic_string<char, char_traits<char>,
      allocator<char> >') for 1st argument
bool hasPrefix(const string & s, const string & prefix);
     ^
1 error generated.
make: *** [mk/patterns.mk:3: src/libmain/shared.o] Error 1
```

I'm on macOS 10.14.6

**Comment by edolstra** [comment]
Fixed in abb8ef619ba2fab3ae16fb5b5430215905bac723.

**Comment by nmattia** [comment]
Confirmed, thanks a bunch!
